### PR TITLE
Add api for departing from an origin

### PR DESCRIPTION
**Pull request by eeyun**
Signed-off-by: Ian Henry <ihenry@chef.io>

### CODE REVIEW

File: components/builder-api/doc/api.raml [hunk 1]
```diff
@@ -878,6 +878,19 @@ securitySchemes:
                       description: Cannot transfer to self or non-members
                     500:
                       description: Internal server error
+        /depart:
+          post:
+            description: Leave an origin you no longer wish to be a member of
+            securedBy: [oauth_2_0]
+            response:
+              204:
+                description: Transfer successful
```
Comment (jeremymv2) [inline components/builder-api/doc/api.raml]: `s/Transfer/Departure/` ?

File: components/builder-api/src/server/resources/origins.rs [hunk 2]
```diff
@@ -1101,6 +1103,49 @@ fn transfer_origin_ownership(req: HttpRequest,
...
+    if !check_origin_member(&req, &origin, session.get_id()).unwrap_or(false) {
+        return HttpResponse::new(StatusCode::UNPROCESSABLE_ENTITY);
+    }
+
+    let conn = match state.db.get_conn().map_err(Error::DbError) {
+        Ok(conn_ref) => conn_ref,
+        Err(err) => return err.into(),
+    };
+
+    debug!(" Departing from origin {}", &origin);
```
Comment (chefsalim) [inline components/builder-api/src/server/resources/origins.rs]: Maybe add the username to the debug output as well, and remove extra leading space?

File: components/builder-api/src/server/resources/origins.rs [hunk 3]
```diff
@@ -1101,6 +1103,49 @@ fn transfer_origin_ownership(req: HttpRequest,
...
+
+    let account_id = match Account::get(&session.get_name(), &*conn).map_err(Error::DieselError) {
```
Comment (chefsalim) [inline components/builder-api/src/server/resources/origins.rs]: I believe you don't need to do this, the session id is the account id.. so you should be able to get it via `session.get_id()`

File: test/builder-api/src/origins.js [hunk 4]
```diff
@@ -257,6 +257,44 @@ describe('Related Origin API functions', function () {
...
+            request.post('/depot/origins/umbrella/depart')
+            .set('Authorization', global.weskerBearer)
+            .expect(204)
+                .end(function (err, res) {
+                    expect(res.text).to.be.empty;
+                    done(err)
+                });
+        });
+    });
+
```
Comment (chefsalim) [inline test/builder-api/src/origins.js]: Add a check that user is no longer a member of the origin after the depart succeeds?

File: components/builder-api/doc/api.raml [hunk 5]
```diff
@@ -878,6 +878,19 @@ securitySchemes:
                       description: Cannot transfer to self or non-members
                     500:
                       description: Internal server error
+        /depart:
+          post:
+            description: Leave an origin you no longer wish to be a member of
+            securedBy: [oauth_2_0]
```
Comment (chefsalim) [inline components/builder-api/doc/api.raml]: Side note - these calls are secured by bearer token, not oauth ... that should be a raml-wide update (separate PR)

Comment (eeyun) [reply components/builder-api/doc/api.raml]: https://github.com/habitat-sh/builder/issues/1294

**Comment by eeyun** [comment]
All feedback addressed

**Pull request merged**